helper/type_mapping_discovery.js

```diff
@@ -0,0 +1,73 @@
+const _ = require('lodash');
+const elasticsearch = require('elasticsearch');
+const peliasConfig = require('pelias-config').generate();
+const logger = require('pelias-logger').get('api:type_mapping_discovery');
+
+/**
+ * This module allows discovery of the sources and layers used
+ * in an existing elasticsearch index.
+ * 
+ * note: this will override any previously configured type mappings.
+ */
+
+const DISCOVERY_QUERY = {
+  requestCache: true,
+  preference: '_replica_first',
```

Comment:
I'd suggest removing this line: unlike say, a traditional SQL database which might have many replicas which are perfect for expensive read-only operations, most Elasticsearch setups do not give different resources to a replica shard vs a primary shard, so it won't have any effect on performance.

Reply:
Yes, actually this line should definitely be removed: `_replica_first` and other similar preference values will [be removed in Elasticsearch 7](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/search-request-preference.html).

There's a big warning at the bottom of that page saying that those preference values didn't do what people expected.

Reply:
I see, the docs say:

>  They do not help to avoid inconsistent results that arise from the use of shards that have different refresh states

I wasn't actually considering that because we use a read-only index, it was more about trying to reduce the impact on the primary nodes, either way, it was only really wishful thinking.

I've removed it via an additional commit